palace.pyx

```diff
@@ -1158,6 +1160,97 @@ cdef class SourceGroup:
         """
         self.impl.destroy()
 
+
+cdef class AuxiliaryEffectSlot:
+    """An effect processor.
+
+    It takes the output mix of zero or more sources,
+    applies DSP for the desired effect (as configured
+    by a given Effect object), then adds to the output mix.
+
+    Parameters
+    ----------
+    context : Optional[Context]
+        The context from which the auxiliary effect slot is to be created.
+        If it is `None`, the object is left uninitialized.
+    """
+    cdef alure.AuxiliaryEffectSlot impl
+
+    def __init__(self, context: Optional[Context]) -> None:
+        if context is None: return
+        self.impl = (<Context> context).impl.create_auxiliary_effect_slot()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]) -> Optional[bool]:
+        self.destroy()
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl < (<AuxiliaryEffectSlot> other).impl
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl <= (<AuxiliaryEffectSlot> other).impl
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl == (<AuxiliaryEffectSlot> other).impl
+
+    def __ne__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl != (<AuxiliaryEffectSlot> other).impl
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl > (<AuxiliaryEffectSlot> other).impl
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl >= (<AuxiliaryEffectSlot> other).impl
+
+    def __bool__(self) -> bool:
+        return <boolean> self.impl
+
+    def set_gain(self, gain: float) -> None:
+        """If set to true, the reverb effect will automatically apply adjustments
```

Comment:
This is too long (limit docstrings to 72 columns, not sure if flake8 will catch this).

Reply:
Where does that count from? If it's from `If...` then there are 70 characters here. I suppose it's from `"""` then.

Reply:
From the start of line.  The convention (see PEP 257) aims to keep the display `help` within 80 columns, and indented blocks in code are indented in `help` doc as well so...